src/azul/service/elasticsearch_service.py

```diff
@@ -434,6 +434,7 @@ def apply_paging(self,
                            _generate_paging_dict can know if there is another
                            page. Use this to prevent a last page that's empty.
         """
+        assert isinstance(peek_ahead, bool), type(peek_ahead)
```

Comment:
I think this would be better placed below, closer to the usage. Assertions are not input validation, they express intent and assumptions.